src/Components/Topics.jsx

```diff
@@ -0,0 +1,39 @@
+import { useEffect, useState } from "react";
+import { getTopics } from "./Api";
+import { Link } from "react-router-dom";
+
+export default function Topics() {
+  const [selectedTopic, setSelectedTopic] = useState(null);
+  const [allTopics, setAllTopics] = useState([]);
+
+  useEffect(() => {
+    getTopics().then((topics) => {
+      setAllTopics(topics);
+    });
+  }, []);
+
+  const handleTopicClick = (topicName) => {
+    setSelectedTopic(topicName);
+  };
+
+  return (
+    <div className="topics">
+      <h2>topics</h2>
+      <ul>
+        {allTopics.map((topic) => {
+          return (
+            <li key={topic.slug}>
+              <Link
+                to={`/articlesByTopic/${topic.slug}`}
+                onClick={() => handleTopicClick(topic.slug)}
+                className={selectedTopic === topic.slug ? "active" : ""}
```

Comment:
I actually quite like what you're doing here by highlighting the selected topic. The only problem is I think it's redundant because when i click on a topic link i will be redirected to a different page altogether and this list of links doesn't actually get seen. So it doesn't matter which topic is selected in this list, the buttons only get seen when there is no selected topic.